service/src/main/java/uk/gov/hmcts/reform/fpl/service/UploadDocumentsService.java

```diff
@@ -0,0 +1,116 @@
+            caseData.getSocialWorkEvidenceTemplateDocument(),
+                caseDataBefore.getSocialWorkEvidenceTemplateDocument());
+        Document thresholdDocument = setUpdatedByAndDateAndTimeForDocuments(
+            caseData.getThresholdDocument(), caseDataBefore.getThresholdDocument());
+        Document checklistDocument = setUpdatedByAndDateAndTimeForDocuments(
+            caseData.getChecklistDocument(), caseDataBefore.getChecklistDocument());
+        Document courtBundleDocument = setUpdatedByAndDateAndTimeForDocuments(
+            caseData.getCourtBundle(), caseDataBefore.getCourtBundle()
+        );
+
+        Map<String, Object> updatedCaseData = new HashMap<>();
+
+        updatedCaseData.put("documents_socialWorkOther", listOfOtherDocs);
+        updatedCaseData.put("documents_socialWorkChronology_document", socialWorkChronologyDocument);
+        updatedCaseData.put("documents_socialWorkStatement_document", socialWorkStatementDocument);
+        updatedCaseData.put("documents_socialWorkAssessment_document", socialWorkAssessmentDocument);
+        updatedCaseData.put("documents_socialWorkCarePlan_document", socialWorkCarePlanDocument);
+        updatedCaseData.put("documents_socialWorkEvidenceTemplate_document", socialWorkEvidenceTemplateDocument);
+        updatedCaseData.put("documents_threshold_document", thresholdDocument);
+        updatedCaseData.put("documents_checklist_document", checklistDocument);
+        updatedCaseData.put("courtBundle", courtBundleDocument);
+
+        return updatedCaseData;
+    }
+
+    public <T extends DocumentMetaData> List<Element<T>> setUpdatedByAndDateAndTimeForDocuments(
+        List<Element<T>> listOfCurrentDocs,
+        List<Element<T>> listOfOldDocs) {
+
+        Predicate<Element<T>> doesNotContainInOldDocs = doc -> listOfOldDocs != null && !listOfOldDocs.contains(doc);
+
+        listOfCurrentDocs.stream()
+            .filter(doesNotContainInOldDocs)
+            .forEach(doc -> findElement(doc.getId(), listOfOldDocs)
+                .ifPresent(e -> {
+                    if (!e.getValue().getTypeOfDocument().equals(doc.getValue().getTypeOfDocument())) {
+                        setUpdatedByAndDateTime(doc);
+                    }
+                })
+            );
+
+        listOfCurrentDocs.stream()
+            .filter(doc -> doc.getValue().getDateTimeUploaded() == null)
+            .forEach(this::setUpdatedByAndDateTime);
+
+        return listOfCurrentDocs;
+    }
+
+    public Document setUpdatedByAndDateAndTimeForDocuments(Document currentDoc, Document oldDoc) {
+
+        if (currentDoc == null) {
+            return null;
+        }
+
+        if ((oldDoc != null && !currentDoc.getTypeOfDocument().equals(oldDoc.getTypeOfDocument()))
+            || currentDoc.getDateTimeUploaded() == null) {
+            return buildDocument(currentDoc);
+        }
+        return null;
+    }
+
+    private Document buildDocument(Document currentDoc) {
```

Comment:
not building, we are setting here. Param should also just be document.

Reply:
method is building document using a builder 😄 . What do you suggest?

Reply:
Fair point but building document is not this method's intention. It's intent is to set both dateTimeUploaded and UploadedBy on a document. `setUploadedByAndDateTimeOnDocument` perhaps?